File: backend/app/schemas/subscription.py
from pydantic import BaseModel
from typing import Any
from datetime import datetime
from uuid import UUID


class PreferencesBase(BaseModel):
    currency_pairs: list[str]
    email_frequency: str = "daily"
    alert_thresholds: dict[str, Any] | None = None
    is_active: bool = True


class PreferencesCreate(PreferencesBase):
    pass


class PreferencesUpdate(BaseModel):
    currency_pairs: list[str] | None = None
    email_frequency: str | None = None
    alert_thresholds: dict[str, Any] | None = None
    is_active: bool | None = None


class PreferencesResponse(PreferencesBase):
    id: UUID
    user_id: UUID
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True

File: frontend/src/types/subscription.ts
export interface UserPreferences {
  id: string
  user_id: string
  currency_pairs: string[]
  email_frequency: 'hourly' | 'daily' | 'weekly'
  alert_thresholds: Record<string, any> | null
  is_active: boolean
  created_at: string
  updated_at: string
}

export interface PreferencesUpdate {
  currency_pairs?: string[]
  email_frequency?: 'hourly' | 'daily' | 'weekly'
  alert_thresholds?: Record<string, any> | null
  is_active?: boolean
}